### PR TITLE
deps: remove iplib in favor of net/netip

### PR DESCRIPTION
**Pull request by kotakanbe**
## Why (motivation for removing this dependency)

- `c-robinson/iplib` is a library for IP address/CIDR manipulation
- Only used in a single function `enumerateHosts()` in `config/tomlloader.go` to expand CIDR notation into individual IP addresses
- Go's stdlib `net/netip` (added in Go 1.18) provides equivalent functionality with `netip.ParsePrefix` and `Addr.Next()`
- iplib is a direct-only dependency — removing it eliminates it completely from go.mod and go.sum

## What (replacement details)

- Rewrote `enumerateHosts()` using `net/netip.ParsePrefix` + `Addr.Next()` iteration
- Preserved IPv4 network/broadcast address stripping for prefix lengths < /31
- Added `TestEnumerateHosts` unit test covering IPv4 (/30, /31, /32), IPv6 (/126, /127, /128), overflow (/32 IPv6), and non-CIDR inputs

### Changed files

| File | Change |
|------|--------|
| `config/tomlloader.go` | Rewrote `enumerateHosts()` from iplib to net/netip (~30 lines), removed iplib import |
| `config/tomlloader_test.go` | Added `TestEnumerateHosts` with 9 test cases |
| `go.mod` / `go.sum` | Removed `c-robinson/iplib` |

## Safety (why this is safe)

- Risk level: **medium** (IP math, but well-scoped to 1 function)
- Behavioral parity verified by existing `TestHosts` (11 test cases covering IPv4/IPv6 CIDR ranges, ignore lists, edge cases)
- New `TestEnumerateHosts` directly tests the rewritten function
- IPv4 network/broadcast stripping matches iplib behavior: strip for prefix < /31, include all for /31-/32 (RFC 3021)

## Test plan

- [x] `TestHosts` — existing 11 test cases pass (behavioral parity)
- [x] `TestEnumerateHosts` — new 9 test cases pass (direct function coverage)
- [x] `go build ./cmd/...` — pass
- [x] `go test ./...` — all pass
- [x] `GOEXPERIMENT=jsonv2 golangci-lint run ./...` — 0 issues

## Review hint (how to review efficiently)

1. **`config/tomlloader.go`** — the core change, ~30 lines. Key logic: `netip.ParsePrefix` → iterate with `Addr.Next()` → strip network/broadcast for IPv4 < /31
2. **`config/tomlloader_test.go`** — new test function, straightforward table-driven tests
3. **`go.mod`** — confirm iplib is removed entirely

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: config/tomlloader.go [hunk 1]
```diff
@@ -193,27 +193,29 @@ func enumerateHosts(host string) ([]string, error) {
...
-			addrs = append(addrs, addr.String())
-		}
+	hostBits := prefix.Addr().BitLen() - prefix.Bits()
+	if hostBits > 63 {
+		return nil, xerrors.Errorf("Failed to enumerate IP address. err: mask bitsize too big")
+	}
+
+	count := uint64(1) << hostBits
+	addrs := make([]string, 0, count)
+	addr := prefix.Masked().Addr()
```
[the remark this answers is not in the document]
Comment (kotakanbe) [reply config/tomlloader.go]: Fixed in c35f046. Changed `count` from `uint64` to `int` so `make([]string, 0, count)` receives the correct type directly. Additionally, the hard upper bound was tightened from 2^63 to 2^16 host bits, which guarantees `count` always fits comfortably in an `int` on any platform.

File: config/tomlloader.go [hunk 2]
```diff
@@ -193,27 +193,29 @@ func enumerateHosts(host string) ([]string, error) {
...
+	for range count {
+		addrs = append(addrs, addr.String())
+		addr = addr.Next()
+	}
```
[the remark this answers is not in the document]
Comment (kotakanbe) [reply config/tomlloader.go:215]: Fixed in c35f046. Tightened the upper bound from 2^63 to 2^16 host bits (`maxHostBits = 16`), so the maximum enumerable range is now a /16 for IPv4 (65,536 addresses) or /112 for IPv6. Wider prefixes like /0 or /8 now return an error immediately instead of attempting a huge allocation. Added test cases for `10.0.0.0/0` and `192.168.0.0/8` to verify.
